### ISSUE TITLE
TypeTree of Vector{Mutable Struct}

### ISSUE THREAD

**Issue by wsmoses** [wsmoses/Enzyme.jl#183]
```
mutable struct MStruct
    val::Float32
end

function lagrangeLeapFrog(domain::Vector{MStruct})
   @inbounds domain[1] = MStruct(0)
   return nothing
end

using Enzyme
Enzyme.API.printtype!(true)

Enzyme.autodiff(lagrangeLeapFrog, Duplicated(MStruct[], MStruct[]))
```

This results in a crash due to the typetree being set up improperly.

**Comment by vchuravy** [comment]
Was this fixed by #184 ?

**Comment by wsmoses** [comment]
Yes
